README.md

```diff
@@ -13,64 +13,99 @@ pip install . # pip install ProtocolDataUnits
-| Additional Features            |                                  |                                      |
-| Pretty Printing                | Yes                              | Via __repr__ method                  |
-| Metadata Support               | Through PDUInfo                  | Yes, via metadata attribute          |
+`ProtocolDataUnits` is a Python toolset for encoding and decoding Protocol Data Units (PDUs). It includes the following features:
+
+- **Base PDU Definition**: Define the structure and format of your PDUs.
+- **PDU Encoding/Decoding**: Encode and decode PDUs with various field types.
+- **Nested PDU Support**: Supports PDUs within PDUs, allowing for complex data structures.
+- **CRC32 Checksum**: Automatically compute and validate CRC32 checksums for data integrity.
+- **Field Encoding/Decoding**: Supports a variety of field types including integers, floats, strings, and arrays.
+- **Byte Order Conversion**: Flexibly handle big-endian and little-endian byte orders.
+- **Metadata Storage**: Store additional metadata within PDUs.
+- **Stream Writing/Reading**: Efficiently write and read PDUs to and from streams.
+- **Variable Length Encoding/Decoding**: Handle fields with variable length data.
+- **Pretty Printing of PDUs**: Generate human-readable representations of PDUs.
+- **PDU Equality based on Fields**: Compare PDUs based on their field values.
+- **Serialization and Deserialization**: Serialize PDU definitions to JSON and deserialize them back to PDU objects.
 
 ## Usage
 
+### Creating PDU Formats
+
+You can create PDU formats using the user-friendly API:
+
 ```python
-from ProtocolDataUnits.pdu import PDU, EthernetFrame
-
-# Create a PDU
-pdu = PDU(type=0x01, duration=500, payload=b'\x01\x02\x03\x04')
-encoded_pdu = pdu.encode()
-decoded_pdu = PDU.decode(encoded_pdu)
-
-# Create an Ethernet Frame
-frame = EthernetFrame(
-    dstaddr=(0x01, 0x02, 0x03, 0x04, 0x05, 0x06),
-    srcaddr=(0x11, 0x12, 0x13, 0x14, 0x15, 0x16),
-    ethtype=0x0800,
-    payload=bytes([0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08])
+from ProtocolDataUnits.pdu import create_pdu_format, PDU
+
+# Create a simple PDU format
+my_pdu_format = create_pdu_format(
+    24, 'big',
+    ('uint8', 'type'),
+    ('float', 'value1'),
+    ('double', 'value2')
+)
+
+# Encode data
+encoded_bytes = my_pdu_format.encode({'type': 7, 'value1': 3.14, 'value2': 6.28})
+print(f"Encoded Bytes: {encoded_bytes}")
+
+# Decode data
+decoded_data = my_pdu_format.decode(encoded_bytes)
+print(f"Decoded Data: {decoded_data}")
+```
+
+### Nested PDU Example
+
+```python
+# Define a nested PDU format
+nested_pdu = create_pdu_format(
+    8, 'big',
+    ('uint8', 'nested_type'),
+    ('uint8', 'nested_value')
+)
+
+# Define the main PDU format containing the nested PDU
+main_pdu = create_pdu_format(
+    16, 'big',
+    ('uint8', 'type'),
+    ('nested_pdu', 'nested', nested_pdu)
```

Comment:
Would `'pdu'` or `'pdu_fragment'` be more descriptive than `'nested_pdu'`?

Reply:
Agreed, and changed all occurrences of the `nested_pdu` to `pdu_fragment` for better clarity, where `nested_pdu` is used in defining fields and change it to `pdu_fragment`.